src/org/labkey/test/tests/SampleSetRemoteAPITest.java

```diff
@@ -181,6 +196,328 @@ public void sampleSetWithMissingValueField() throws IOException, CommandExceptio
         dgen.deleteDomain(createDefaultConnection(true));
     }
 
+    @Test
+    public void insertSamplesOverSaveBatchAPI() throws Exception
+    {
+        String missingValueTable = "mvAssaySamples";
+        String missingValueSamplesFolder = "mvAssaySamplesFolder";
+        _containerHelper.createSubfolder(getProjectName(), missingValueSamplesFolder);
+
+
+        // create assay
+        String assaySubfolder = "TestAssayFolder";
+        String assayName = "AssayForSampleDerivation";
+        generateAssay(assaySubfolder, assayName)
+                .addDataField("SampleName", "SampleName", FieldDefinition.ColumnType.String)
+                .addDataField("SampleVolume", "SampleVolume", FieldDefinition.ColumnType.Double)
+                .saveAndClose();
+        insertAssayData(assayName, assaySubfolder, new FieldDefinition.LookupInfo(getProjectName() + "/" + assaySubfolder,
+                "assay.General.AssayForSampleDerivation_assay", "Runs"));
+
+        // get the sampleID, it will be needed later
+        waitAndClickAndWait(Locator.linkWithText("Assay List"));
+        DataRegionTable assayList = DataRegionTable.DataRegion(getDriver()).withName("AssayList").waitFor();
+        CustomizeView customizeView = assayList.openCustomizeGrid();
+        customizeView.showHiddenItems();
+        customizeView.addColumn("ROWID");
+        customizeView.clickSave().save();   // make this the default view
+        String assayIdStringValue = assayList.getRowDataAsMap(0).get("RowId");
+
+        // create sampleset
+        navigateToFolder(getProjectName(), missingValueSamplesFolder);
+        TestDataGenerator dgen = new TestDataGenerator("exp.materials", missingValueTable, getProjectName() + "/" + missingValueSamplesFolder)
+                .withColumnSet(List.of(
+                        TestDataGenerator.simpleFieldDef("name", FieldDefinition.ColumnType.String),
+                        TestDataGenerator.simpleFieldDef("mvStringData", FieldDefinition.ColumnType.String)
+                                .setMvEnabled(true).setLabel("MV Field"),
+                        TestDataGenerator.simpleFieldDef("volume", FieldDefinition.ColumnType.Double)
+                ));
+        dgen.createDomain(createDefaultConnection(true), "SampleSet");
+        dgen.addCustomRow(Map.of("name", "1st", "volume", 13.5));
+        dgen.addCustomRow(Map.of("name", "2nd", "volume", 15.5));
+        dgen.addCustomRow(Map.of("name", "3rd", "volume", 16.5));
+        dgen.addCustomRow(Map.of("name", "4th", "volume", 17.5));
+        dgen.insertRows(createDefaultConnection(true), dgen.getRows());
```

Comment:
This is missing a field to create a missing value. If the mvStringData (MV Field) field is to be a missing value field you need to se the mvstringdata_mvindicator field as well. The *_mvindicator field is the one that indicates what the actual missing value is, without setting it you just have an empty field.

Also be aware that the case of the *_mvindicator field is different in MSSQL (_MVIndicator) and Postgres (_mvindicator).

Reply:
I thought the `*_mvindicator` field is automatically created. Or maybe it's just automatic in the UI because the GWT designer knows to create it when that checkbox is checked.

Reply:
In the UI when you edit/insert a missing value field there is a drop down that sets the *_mvindicator field. If you don't set it the value in the field is not considered missing.

Reply:
Gotcha. I misunderstood. I was thinking about creating the field, not setting it.